molgenis-security/src/main/java/org/molgenis/security/WebAppSecurityInitializer.java

```diff
@@ -1,5 +1,23 @@
 package org.molgenis.security;
 
+import static javax.servlet.DispatcherType.ASYNC;
+import static javax.servlet.DispatcherType.FORWARD;
+import static javax.servlet.DispatcherType.REQUEST;
+
+import java.util.EnumSet;
+import javax.servlet.FilterRegistration.Dynamic;
+import javax.servlet.ServletContext;
 import org.springframework.security.web.context.AbstractSecurityWebApplicationInitializer;
+import org.springframework.web.filter.ForwardedHeaderFilter;
+
+public class WebAppSecurityInitializer extends AbstractSecurityWebApplicationInitializer {
 
-public class WebAppSecurityInitializer extends AbstractSecurityWebApplicationInitializer {}
+  @Override
+  protected void beforeSpringSecurityFilterChain(ServletContext servletContext) {
+    // add filters
+    Dynamic forwardedHeaderFilter =
+        servletContext.addFilter("forwardedHeaderFilter", ForwardedHeaderFilter.class);
+    forwardedHeaderFilter.setAsyncSupported(true);
+    forwardedHeaderFilter.addMappingForUrlPatterns(EnumSet.of(REQUEST, FORWARD, ASYNC), false, "*");
```

Comment:
Why did you add 'FORWARD'?

Reply:
Oops, that's a leftover from previous attempt to fix it. I noticed the filter wasn't getting executed and thought perhaps the inner workings of the OIDC filter included a forward request.

I just now checked how spring boot registers the filter: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/servlet/ServletWebServerFactoryAutoConfiguration.java

They use request, async, error. So forward not needed, removing it.
Shall we add error to match what spring boot does?

Reply:
Good idea.